tensorflow/contrib/learn/python/learn/datasets/base.py

```diff
@@ -144,3 +144,34 @@ def maybe_download(filename, work_directory, source_url):
         size = f.size()
       print('Successfully downloaded', filename, size, 'bytes.')
   return filepath
+
+def split_into_train_and_test(dataset, test_ratio=0.3, shuffle=False):
```

Comment:
I have not touched the file in the `_sklearn` -- There are some compatibility issues, namely:
-- `_sklearn._train_test_split` expects variable arguments, and uses `.take()` to extract the values. Current implementation expects `base.Dataset` as an imput. If I replace it with the implementation in this PR, it will break any other codes.
-- `_sklearn._train_test_split` returns tuples, while the `contrib.learn` methods return either `base.Dataset` or `base.Datasets`, which are not iterable per se.

I could completely nuke the `_sklearn` method, but it is potentially dangerous. Please, advise